nms/app/views/domain-proxy/CbsdEdit.js

```diff
@@ -0,0 +1,425 @@
+  fccId: string,
+  userId: string,
+  minPower: number,
+  maxPower: number,
+  numberOfAntennas: number,
+  antennaGain: number,
+  bandwidthMhz: BandwidthEnum,
+  frequenciesMhz: number[],
+};
+
+const convertToCbsdFormData = (cbsd?: cbsd): CbsdFormData => {
+  return {
+    serialNumber: cbsd?.serial_number || '',
+    fccId: cbsd?.fcc_id || '',
+    userId: cbsd?.user_id || '',
+    minPower: cbsd?.capabilities?.min_power || 0,
+    maxPower: cbsd?.capabilities?.max_power || 0,
+    numberOfAntennas: cbsd?.capabilities?.number_of_antennas || 0,
+    antennaGain: cbsd?.capabilities?.antenna_gain || 0,
+    bandwidthMhz: cbsd?.frequency_preferences?.bandwidth_mhz || 5,
+    frequenciesMhz: cbsd?.frequency_preferences?.frequencies_mhz || [0],
+  }
+}
+
+export function CbsdAddEditDialog(props: DialogProps) {
+  const enqueueSnackbar = useEnqueueSnackbar();
+
+  const classes = useStyles();
+
+  const ctx = useContext(CbsdContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [formErrors, setFormErrors] = useState < string[] > ([]);
+  const [cbsdFormData, setCbsdFormData] = useState < CbsdFormData > (convertToCbsdFormData());
+
+  useEffect(() => {
+    const newValue = convertToCbsdFormData(props?.cbsd);
+    setCbsdFormData(newValue);
+  }, [props.open, props.cbsd]);
+
+  useEffect(() => {
+    return () => setFormErrors([]);
+  }, []);
+
+  const onSave = async () => {
+    try {
+      setIsLoading(true);
+
+      const cbsdData: mutable_cbsd = {
+        capabilities: {
+          min_power: parseInt(cbsdFormData.minPower),
+          max_power: parseInt(cbsdFormData.maxPower),
+          number_of_antennas: parseInt(cbsdFormData.numberOfAntennas),
+          antenna_gain: parseInt(cbsdFormData.antennaGain),
+        },
+        frequency_preferences: {
+          bandwidth_mhz: cbsdFormData.bandwidthMhz,
+          frequencies_mhz: cbsdFormData.frequenciesMhz.map((value) => parseInt(value)),
+        },
+        fcc_id: cbsdFormData.fccId,
+        serial_number: cbsdFormData.serialNumber,
+        user_id: cbsdFormData.userId,
+      };
+      if (!props.cbsd) {
+        await ctx.create(cbsdData);
+      } else {
+        await ctx.update(props.cbsd.id, cbsdData);
+      }
+
+      props.onClose();
+      enqueueSnackbar('CBSD saved successfully', {
+        variant: 'success',
+      });
+    } catch (e) {
+      if (e.response?.data?.errors?.length) {
+        const getErrorMessages = (errors): string[] => {
+          return errors
+            .map((item) => {
+              return item.errors
+                ? getErrorMessages(item.errors)
+                : item.message;
+            })
+            .flat();
+        }
+
+        const validationErrors = getErrorMessages(e.response.data.errors);
+        setFormErrors(validationErrors);
+      } else {
+        setFormErrors([e.response?.data?.message ?? e.message]);
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
```

Comment:
The NMS does have a very bad test test coverage. I want to change this.
Please add some basic test e.g. for editing and creating a CBSD.